### PR TITLE
kubeadm: Use GetGenericImage for kube-dns

### PR DESCRIPTION
**Pull request by rosti**
**What this PR does / why we need it**:

As kube-dns transitioned to fat manifests (in #68430), it's no longer required to use arch
suffixed images. This change makes use of fat manifests for kube-dns and
removes the last few calls to the GetGenericArchImage function, thus removing
GetGenericArchImage too.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
NONE

**Special notes for your reviewer**:

Since this PR removes GetGenericArchImage, I'm putting it on hold until #59664 get merged. This is needed, because since #66960, etcd is also required to have a fat manifest (which currently it does not).

/hold
/cc @kubernetes/sig-cluster-lifecycle-pr-reviews
/area kubeadm
/kind cleanup
/assign @fabriziopandini 
/assign @timothysc
/assign @dims
/cc @mkumatag

**Release note**:

```release-note
kubeadm now uses fat manifests for the kube-dns images
```

### CODE REVIEW

**Comment by neolit123** [comment]
/priority critical-urgent
adding this priority because the PR will complete our full span of manifest list support in 1.12.

**Comment by dims** [comment]
/milestone v1.12

(will defer to @timothysc and others to clear the milestone if not appropriate / or it is too late)

**Review by dixudx — APPROVED** [review]
/lgtm

**Review by timothysc — COMMENTED** [review]
/approve 
/hold cancel

**Pull request merged**
